library/SimplePie/Item.php

```diff
@@ -2862,11 +2862,11 @@ public function get_enclosures()
 					$width = null;
 
 					$url = $this->sanitize($enclosure[0]['attribs']['']['url'], SIMPLEPIE_CONSTRUCT_IRI, $this->get_base($enclosure[0]));
-					if (isset($enclosure[0]['attribs']['']['type']))
+					if (!empty($enclosure[0]['attribs']['']['type']))
 					{
 						$type = $this->sanitize($enclosure[0]['attribs']['']['type'], SIMPLEPIE_CONSTRUCT_TEXT);
 					}
-					if (isset($enclosure[0]['attribs']['']['length']))
+					if (!empty($enclosure[0]['attribs']['']['length']))
 					{
 						$length = ceil($enclosure[0]['attribs']['']['length']);
```

Comment:
Here was the actual crash with PHP8+

Reply:
It will still crash when invalid `enclosure="foo"` attribute is provided. Perhaps rather than checking emptiness, we should check if the value is a number.

Reply:
@jtojnar Right, this should do the job https://github.com/simplepie/simplepie/pull/670/commits/3000a70c2692988f67ea346e0dc3d4ae8289b724

Reply:
`intval()` will cast any wrong value to 0

Reply:
With that approach, the behaviour is unchanged compared to PHP7-

Reply:
That should work (with slightly different semantics for numbers with decimal point) but now the `ceil` is not longer necessary.

Reply:
Ah, right, on mobile now, but will simplify tomorrow

Reply:
Otherwise `floatval()` could be an option to keep `ceil()`

Reply:
`intval()` is probably better, since length and fileSize are both expressed in bytes anyway